rfc/0052-sdk3-full-text-search.md

```diff
@@ -606,18 +606,154 @@ interface SearchRowLocation {
 
 ### SearchFacetResult
 
-An individual facet result has both metadata and details, as each facet can define ranges into which results are categorized.
+The `facets` part of the result, a sibling to `hits`, contains information relative to the facets the user asked for. If no `facets` section was provided in the query, this section is omitted.
+
+Note that the whole `facets` section **is a single JSON object**. The `name` of each facet, which is defined at query time by the user, is represented in the attribute names.
+
+An individual facet result has both metadata and details, as each facet can define ranges into which results are categorized. The JSON representation of a single facet's results looks like this:
+
+```
+SearchFacetResult {
+    "field": string
+    "total": long
+    "missing": long
+    "other": long
+    "terms": [ {TermRange}, ... ] <1>
+    "numeric_ranges": [ {NumericRange}, ...] <1>
+    "date_ranges": [ {DateRange}, ... ] <1>
+}
+```
+(1) all mutually exclusive (`terms`, `numeric_ranges` and `date_ranges` cannot be combined).
```

Comment:
maybe add a note that languages which support inheritance usually will have the upper fields in the SearchFacetResult and then the three others extend from it and add one of the 3  for the child class

Reply:
This is covered below and this is meant to be a representation of that the JSON format is.